test/SILOptimizer/definite_init_closures.swift

```diff
@@ -0,0 +1,50 @@
+// RUN: %target-swift-frontend -emit-sil %s -o /dev/null
+
+// Test boolean operators with implicit closures
+
+struct Simple {
+  let x: Bool
+  let y: Bool
+
+  init() {
+    x = false
+    y = false || x
+  }
+}
+
+struct NestedClosures {
```

Comment:
Please add a test for a generic struct or a struct with an existential field to test the address only case

Reply:
done